src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,185 +1,96 @@
         qty,
       };
-      carts.push(objInserted)
-    }else{
+      carts.push(objInserted);
+    } else {
       var obj = carts[position];
-      var objInserted = {...obj, qty: obj.qty+this.state.qty};
+      var objInserted = { ...obj, qty: obj.qty + this.state.qty };
       carts[position] = objInserted;
     }
-    this.setState({
-      showModal: false,
-      qty: 1,
-      selectedProduct: {},
-    },()=>{
-      this.recalculateGrandTotal();
-      console.log(`cart`, this.state.cart)
-    })
+    setShowModalProduct(false);
+    setQuantity(1);
+    setSelectedProduct({});
+    recalculateGrandTotal();
   }
 
-  recalculateGrandTotal = () => {
-    var grandTotal = 0.0, itemTotal = 0;
-    
-    this.state.cart.map((v) => {
-      grandTotal += v.qty*v.price;
-      itemTotal += v.qty;
-    })
-
-    this.setState({
-      grandTotal,
-      itemTotal,
-    })
-  }
+  function recalculateGrandTotal() {
+    var grandTotal = 0.0,
+      itemTotal = 0;
 
-  onChangeQty(value){
-    this.setState({
-      qty: value
-    })
-  }
+    cart.map((v) => {
+      grandTotal += v.qty * v.price;
+      itemTotal += v.qty;
+    });
 
-  renderModalProduct(){
-    const Context = styled.div`
-      display: flex;
-      flex-direction: row;
-    `;
-
-    const Image = styled.img`
-      height: 250px;
-    `;
-
-    const ProductInfo = styled.div`
-      margin: 20px;
-      padding: 10px;
-    `;
-
-    const Description = styled.p`
-      font-size: 12px;
-      display: block;
-    `;
-
-    return(
-      <Modal
-        centered
-        title="Selected Product"
-        visible={this.state.showModal}
-        onOk={this.handleOk}
-        onCancel={this.handleCancel}
-        footer={[
-          <Button key="back" onClick={this.addtoCartHandler}>
-            Add to Cart
-          </Button>,
-          <Button
-            key="link"
-            type="primary"
-            onClick={this.handleCancel}
-          >
-            Cancel
-          </Button>,
-        ]}>
-        <Context>
-          <Image src={this.state.selectedProduct.image}
-              style={{ height: "250px" }}/>
-          <ProductInfo>
-            <h2><strong>{this.state.selectedProduct.title}</strong></h2>
-            <Description>
-              {this.state.selectedProduct.description}<br/>
-              <strong>${this.state.selectedProduct.price}</strong>
-            </Description>
-            <InputNumber min={1} defaultValue={1} onChange={(value) => {console.log("Number Value :",value); this.onChangeQty(value)}}/>
-          </ProductInfo>
-        </Context>
-        
-      </Modal>
-    )
+    setGrandtotal(grandTotal);
+    setItemTotal(itemTotal);
   }
 
-  handleOkCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  handleCancelCart = () => {
-    this.setState({ showModalCart: false });
-  };
-
-  render() {
-    const Wrapper = styled.div`
-      height: 100%;
-      width: 100%;
-    `;
+  const Wrapper = styled.div`
```

Comment:
Don't define styled components inside React components. That's bad for performance.